container.go

```diff
@@ -410,8 +411,16 @@ func (c *ContainerRequest) ShouldKeepBuiltImage() bool {
 	return c.FromDockerfile.KeepImage
 }
 
-func (c *ContainerRequest) ShouldPrintBuildLog() bool {
-	return c.FromDockerfile.PrintBuildLog
+func (c *ContainerRequest) BuildLogWriter() io.Writer {
+	if c.FromDockerfile.BuildLogWriter == nil {
+		c.FromDockerfile.BuildLogWriter = os.Stderr
+	}
```

Comment:
question: should we try to maintain current behaviour so default to discard?

Reply:
At the moment, the current behavior maintains because:
- PrintBuildLog is true and BuildLogWriter is default -> os.Stderr
- PrintBuildLog is false and BuildLogWriter is default -> io.Discard

However, it is possible to maintain compatibility in another way:
```go
if c.FromDockerfile.BuildLogWriter != nil {
	return c.FromDockerfile.BuildLogWriter
}
if c.FromDockerfile.PrintBuildLog {
	c.FromDockerfile.BuildLogWriter = os.Stderr
} else {
	c.FromDockerfile.BuildLogWriter = io.Discard
}
return c.FromDockerfile.BuildLogWriter
```

This way might be better because we can mark `PrintBuildLog` as deprecated field.